server/service/client.go

```diff
@@ -1621,10 +1624,19 @@ func (c *Client) DoGitOps(
 
 		// Labels
 		if config.Labels == nil || len(config.Labels) > 0 {
-			err = c.doGitOpsLabels(config, logFn, dryRun)
+			labelsToDelete, err := c.doGitOpsLabels(config, logFn, dryRun)
 			if err != nil {
-				return nil, err
+				return nil, nil, err
 			}
+			postOps = append(postOps, func() error {
+				for _, labelToDelete := range labelsToDelete {
+					logFn("[-] deleting label '%s'\n", labelToDelete)
+					if err := c.DeleteLabel(labelToDelete); err != nil {
+						return err
+					}
+				}
+				return nil
+			})
```

Comment:
Returning the labels to delete from `DoGitopsLabels` makes perfect sense to me.  Adding a new `postOps` mechanism seems a little high-touch.  Would adding a "delete labels" step (`DoGitOpsDeleteLabels()?`) to the end of `DoGitOps()` not be sufficient?

Reply:
Sure, I'll give it a try.

I followed this approach because there are multiple calls to `DoGitOps(...)`  and its not clear to me if they are mutually exclusive and, if they are not, one way to delete labels at 'the end of everything' is to do something like this.

Reply:
Actually -- I see your point.  The `DoGitOps()` run for the global settings goes first, so if all we did was wait until the end of _that_ run to delete the labels, we'd still have the same issue.  We need it to run after _all_ the teams have done their business 🤔 .  And even though labels are all global right now, having team-scoped labels has been discussed, in which case having each team add a cleanup step would be helpful.